tests/docs/pytest.voq_chassis.md

```diff
@@ -0,0 +1,299 @@
+
+Lets consider a chassis with 2 linecards (node1, node2) and a single supervisor card (supervisor1).
+
+# 4.1 Ansible inventory
+
+We define each linecard of a SONiC chassis in the inventory file as a single AnsibleHost. The only change required is that for the linecards, you would add 'num_asics' field, and for supervisor card you would specify the 'type' field as 'supervisor'
+
+```
+sonic:
+  hosts:
+    node1:
+      ansible_host: 10.10.10.1
+      num_asics: 1
+    node2:
+      ansible_host: 10.10.10.2
+      num_asics: 1
+    supervisor1:
+      ansible_host: 10.10.10.3
+      type: supervisor
+      num_asics: 1
+```
+
+# 4.2 connection graph
+
+In the connection graph, you would define each linecard (including supervisor card) and all the links between the linecards of the SONiC chassis and the fanout switch(es).
+
+```
+node1,Ethernet0,fanout1,Ethernet0,100000,100,Access
+node1,Ethernet1,fanout1,Ethernet4,100000,101,Access
+.
+.
+node2,Ethernet0,fanout2,Ethernet0,100000,100,Access
+.
+.
+```
+
+# 4.3 testbed.csv
+
+In testbed.csv, you define the SONiC chassis as a list of duts
+
+<pre>
+# conf-name,group-name,topo,ptf_image_name,ptf,ptf_ip,ptf_ipv6,server,vm_base,dut,comment
+chassis1-t1,vms_5,t1-chassis,docker-ptf,ptf_vms_6,10.250.5.188/24,,server_5,VM0500,<b>[node1;node2;supervisor1]</b>, OC tests for chassis1
+
+</pre>
+
+# 4.4 topology file
+
+We use the multi-dut approach in defining the topology file, where the VM offsets have the format '<dut_index>.<dut_port>@<ptf_index>'. This format was introduced to support [Dual Tor](https://github.com/Azure/sonic-mgmt/pull/2333).  
+
+```
+topology:
+  VMs:
+    VM0100:
+      vlans:
+        - "0.0@0"
+      vm_offset: 0
+    VM0101:
+      vlans:
+        - "0.1@1"
+      vm_offset: 1
+    .
+    .
+    VM0132:
+      vlans:
+        - "1.0@32"
+      vm_offset: 32
+    VM0133:
+      vlans:
+        - "1.1@33"
+      vm_offset: 33
+    .
+    .
+```
+
+# 5 Test case changes
+
+We have outline two approaches above of how to adapt existing tests to be able to run on SONiC chassis 
+- parameterizing dut_index
+- duthosts
+
+Whether you we use the parameterizing approach, or duthosts approach depends upon the test case, where one approach might be easier to adapt the test case for a SONiC chassis. For example, 
+- platform tests might be easier to modify using the parameterizing approach,
+- tests like test_dip_sip if extended to test traffic across multiple asics (same or different linecards) and using ptf's would make sense to run with 'duthosts' approach.
+
+Below, we take the example test_bgp_facts modified with both approaches as an example.
+
+## 5.1 parameterizing dut_index
+
+- Modify definition to replace 'duthost' with 'duthosts', and include 'dut_index' and 'asic_index'.
+- Get duthost based on dut_index from duthosts
+- Call ansible modules 'bgp_facts' and config_facts' using the namespace arguements. 
+
+<pre>
+def test_bgp_facts(<b>duthosts, dut_index, asic_index</b>):
```

Comment:
Fixtures like `dut_index` and `asic_index` have been updated to `enum_dut_hostname` and `enum_asic_index`. Please update occurrences of the deprecated fixture names accordingly.